core/docker/bin/health-check

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function get_property() {
+    grep "^$1=" "$2" | cut -d'=' -f2
+}
+
+scheme=http
+port=8080
+
+config=/etc/trino/config.properties
+if [ "$(get_property 'http-server.https.enabled' "$config")" == true ]; then
+    scheme=https
+    port=$(get_property http-server.https.port "$config")
+    # disable certificate verification in curl, in case a self-signed certificate is being used
+    set -- "$@" --insecure
```

Comment:
This seems like an extremely obscure or obfuscated way to achieve the result. I feel like I'm fairly proficient in bash, but even after reading the [set documentation](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html), I still had to puzzle over what this is doing.

Adding to the confusion is that this script is called with no arguments by the `HEALTHCHECK` command, so `$@` will be empty. If the intent is that we can call the healthcheck script with extra arguments for curl, then that should be documented explicitly as a feature.

I would either

1. Always add `--insecure` since it doesn't matter in the HTTP case
2. Add an explicit `curl_opts` variable that you set here

Reply:
I agree. I was too eager to allow passing custom args to curl, just in case someone would like to use this script for a custom Kubernetes probe, but I'd rather keep this simple now and add it later when requested. I removed using `"$@"` and used long names for all curl flags to make it more readable.